app/main.py

```diff
@@ -1,2 +1,86 @@
+from typing import Optional, Any
+
+
+class Node:
+    def __init__(self, key: Any, hash_value: int, value: Any) -> None:
+        self.key = key
+        self.hash_value = hash_value
+        self.value = value
+        self.next: Optional[Node] = None
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            initial_capacity: int = 16,
+            load_factor: float = 0.75
+    ) -> None:
+        self.capacity: int = initial_capacity
+        self.load_factor: float = load_factor
+        self.size: int = 0
+        self.table: list[Optional[Node]] = [None] * initial_capacity
+
+    def __setitem__(
+            self,
+            key: Any,
```

Comment:
`key` should be not any, but hashable. Indicate this in the type annotation in all places.